File: src/python/generator/lib/dbautoinc/db_auto_inc.py
import mysql.connector
import time
import sys

class DbAutoIncGenerator:
    def __init__(self):
        self.connection = None
        
        config = {
            'host': 'proxysql',
            'port': 6033,
            'user': 'root',
            'password': 'rootpassword',
            'database': 'uuid_db',
            'autocommit': True
        }

        for i in range(30):
            try:
                self.connection = mysql.connector.connect(**config)
                print("Successfully connected to ProxySQL.", flush=True)
                break
            except Exception as e:
                print("Waiting for database connection...", flush=True)
                time.sleep(2)

        if not self.connection:
            print("Database connection failed after retries.", file=sys.stderr)
            sys.exit(1)

    def next_id_string(self):
        if not self.connection:
            print("Database connection not initialized.", file=sys.stderr)
            return ""

        try:
            cursor = self.connection.cursor()
            cursor.execute("REPLACE INTO tickets64 (stub) VALUES ('a')")
            insert_id = cursor.lastrowid
            cursor.close()
            return str(insert_id)
        except Exception as e:
            print(f"Error executing query: {e}", file=sys.stderr)
            return ""

File: src/node/generator/lib/dbautoinc/db_auto_inc.js
const mysql = require('mysql2/promise');

class DbAutoIncGenerator {
    constructor() {
        this.pool = null;
    }

    async init() {
        const config = {
            host: 'proxysql',
            port: 6033,
            user: 'root',
            password: 'rootpassword',
            database: 'uuid_db',
            waitForConnections: true,
            connectionLimit: 10,
            queueLimit: 0
        };

        for (let i = 0; i < 30; i++) {
            try {
                this.pool = mysql.createPool(config);
                const connection = await this.pool.getConnection();
                connection.release();
                console.log("Successfully connected to ProxySQL.");
                return;
            } catch (err) {
                console.log("Waiting for database connection...");
                await new Promise(resolve => setTimeout(resolve, 2000));
            }
        }

        console.error("Database connection failed after retries.");
        process.exit(1);
    }

    async nextIdString() {
        if (!this.pool) {
            console.error("Database pool not initialized.");
            return "";
        }

        try {
            const [result] = await this.pool.execute("REPLACE INTO tickets64 (stub) VALUES ('a')");
            return result.insertId.toString();
        } catch (err) {
            console.error("Error executing query:", err);
            return "";
        }
    }
}

module.exports = DbAutoIncGenerator;
